client/app/bundles/course/user-invitations/pages/InviteUsers/index.tsx

```diff
@@ -0,0 +1,106 @@
+import { FC, useEffect, useState } from 'react';
+import { defineMessages, injectIntl, WrappedComponentProps } from 'react-intl';
+import { toast } from 'react-toastify';
+import { Box, Grid, Typography } from '@mui/material';
+import { useDispatch, useSelector } from 'react-redux';
+import { AppDispatch, AppState } from 'types/store';
+import LoadingIndicator from 'lib/components/LoadingIndicator';
+import PageHeader from 'lib/components/pages/PageHeader';
+import { InvitationResult } from 'types/course/userInvitations';
+import {
+  getManageCourseUsersSharedData,
+  getManageCourseUserPermissions,
+} from '../../selectors';
+import UserManagementTabs from '../../../users/components/navigation/UserManagementTabs';
+import { fetchInvitations } from '../../operations';
+import RegistrationCodeButton from '../../components/buttons/RegistrationCodeButton';
+import UploadFileButton from '../../components/buttons/UploadFileButton';
+import IndividualInviteForm from '../../components/forms/IndividualInviteForm';
+import InvitationResultDialog from '../../components/misc/InvitationResultDialog';
+
+type Props = WrappedComponentProps;
+
+const translations = defineMessages({
+  manageUsersHeader: {
+    id: 'course.users.manage.header',
+    defaultMessage: 'Manage Users',
+  },
+  inviteUsersHeader: {
+    id: 'course.users.userInvitations.header',
+    defaultMessage: 'Invite Users',
+  },
+  fetchInvitationsFailure: {
+    id: 'course.users.userInvitations.fetch.failure',
+    defaultMessage: 'Unable to fetch invitations',
+  },
+});
+
+const InviteUsers: FC<Props> = (props) => {
+  const { intl } = props;
+  const [isLoading, setIsLoading] = useState(true);
+  const [showInvitationResultDialog, setShowInvitationResultDialog] =
+    useState(false);
+  const [invitationResult, setInvitationResult] = useState({});
+  const permissions = useSelector((state: AppState) =>
+    getManageCourseUserPermissions(state),
+  );
+  const sharedData = useSelector((state: AppState) =>
+    getManageCourseUsersSharedData(state),
+  );
+
+  const dispatch = useDispatch<AppDispatch>();
+
+  useEffect(() => {
+    dispatch(fetchInvitations())
```

Comment:
passing additional params is fine.. For course with large number of users, it is slow to then load this page..

also i think sharedData can be selected from inside the UserManagementTabs component? Dont need to select here and pass it since sharedData is not used anywhere else right? (same for permissions). Also check for this issue in other files as well.

Reply:
Ok, I have refactored to specify what to fetch from params